Document multi file upload (#3860)
hammerandchisel/discord-api-docs: docs/Reference.md

```

Ensure you use the proper content type (`image/jpeg`, `image/png`, `image/gif`) that matches the image data being provided.

## Uploading Files

> When creating or editing a message, it is possible to attach files to the message. To add a file to the request, the standard `application/json` body must be replaced by a `multipart/form-data` body. The otherwise json body can instead be provided using a special `payload_json` parameter in addition to a number of `file` parameters.

Reviewer: I think each time we refer to it, it would be good to standardize on `file[n]`

The file upload limit applies to the entire request, not individual files in a request. This limit depends on the **Boost Tier** of a Guild and is 8 MiB by default.

> All `file` parameters must include a valid `Content-Disposition` subpart header with a `filename` and unique `name` parameter. Each file parameter must be uniquely named in the format `file[n]` such as `file0`, `file1`, or simply `file`. The suffixed index can be used as a *pseudo-snowflake* in the `attachments` json parameter that is supplied in `payload_json` to add a `description`.

Author: ```suggestion
All `file` parameters must include a valid `Content-Disposition` subpart header with a `filename` and unique `name` parameter. Each file parameter must be uniquely named in the format `file[n]` such as `file0`, `file1`, or `file42`. The suffixed index `n` is the *snowflake placeholder* for the `attachments` json parameter that is supplied in `payload_json`.
```

> All `file` parameters must include a valid `Content-Disposition` subpart header with a `filename` and unique `name` parameter. Each file parameter must be uniquely named in the format `file[n]` such as `file0`, `file1`, or `file42`. The suffixed index `n` is the *snowflake placeholder* for the `attachments` json parameter that is supplied in `payload_json` (or [Callback Data Payloads](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING/interaction-response-object-interaction-callback-data-structure)).

Reviewer: The brackets are included - `file[0]`, `file[1]`, `file[42]`. What do you think about swapping this paragraph with the one above it, so it goes naming -> limits?

Images can also be referenced in embeds using the `attachments://filename` URL. An example payload is provided below.

### Editing Attachments

All files added to a request, as described above, will be appended to the message in a `PATCH` request. The `attachments` json parameter has a special behavior for edit, as it is used for both removing attachments from the message as well as adding descriptions for new attachments added by the request.

> The `attachments` json parameter lists all files that should be **retained** on the message after the edit request. Any files that are not listed will be removed.

Author: ```suggestion
The `attachments` json parameter lists all files that should be attached to the message after the edit, including all new files added and the respective snowflake placeholders. To remove attachments, simply exclude them from this list.
```

This example demonstrates usage of the endpoint *without* `payload_json`.

```
--boundary
Content-Disposition: form-data; name="content"

Hello, World!
--boundary
Content-Disposition: form-data; name="tts"

true
--boundary--
```

This example demonstrates usage of the endpoint *with* `payload_json` and all content fields (`content`, `embeds`, `file`) set.

```
--boundary
Content-Disposition: form-data; name="payload_json"
Content-Type: application/json

{
  "content": "Hello, World!",
  "embeds": [{
    "title": "Hello, Embed!",
    "description": "This is an embedded message.",
    "thumbnail": {
      "url": "attachment://myfilename.png"
    },
    "image": {
      "url": "attachment://mygif.gif"
    }
  }],
  "message_reference": {
    "message_id": "233648473390448641"
  },
  "attachments": [{
      "id": 0,
      "description": "Image of a cute little cat",
      "filename": "file.png"
  }, {
      "id": 1,
      "description": "Rickroll gif",
      "filename": "mygif.gif"
  }]
}
--boundary
Content-Disposition: form-data; name="file0"; filename="file.png"
Content-Type: image/png

[image bytes]
--boundary

> Content-Disposition: form-data; name="file1"; filename="mygif.gif"

Reviewer: Same callout here, `file[1]`
